src/librustc_driver/driver.rs

```diff
@@ -592,7 +597,7 @@ pub fn phase_2_configure_and_expand<'a, F>(sess: &Session,
                                    addl_plugins.take().unwrap())
     });
 
-    let mut registry = Registry::new(sess, &krate);
+    registry.krate_span = krate.span;
```

Comment:
Do we need to set `registry.krate_span` here? I think the only way that `registry.krate_span != krate.span` at the beginning of this function is if a rogue compiler controller changed the span for some reason.